### PR TITLE
[delta-standalone] Implement actions iterator from `VersionLog` by JAVA

### PR DESCRIPTION
**Pull request by 6a0juu**
Currently VersionLog.getActions returns a list of actions. We would like Iterator to avoid memory pressure.

### CODE REVIEW

File: standalone/src/main/java/io/delta/standalone/VersionLog.java [hunk 1]
```diff
@@ -29,11 +37,19 @@
 public final class VersionLog {
     private final long version;
 
-    @Nonnull
-    private final List<Action> actions;
+    private List<Action> actions;
+
+    private final Supplier<CloseableIterator<String>> actionsSupplier;
+
+    public VersionLog(long version, @Nonnull Supplier<CloseableIterator<String>> actionsSupplier) {
```
[the remark this answers is not in the document]
Comment (scottsand-db) [reply standalone/src/main/java/io/delta/standalone/VersionLog.java:44]: options 1 and 2 are the same

[the remark this answers is not in the document]
Comment (6a0juu) [reply standalone/src/main/java/io/delta/standalone/VersionLog.java:44]: > options 1 and 2 are the same

Updated. 

And There is another issue in [Github CI](https://github.com/delta-io/connectors/runs/6715736289?check_suite_focus=true#step:8:133). Scala 2.11 doesn't support transform JAVA `Supplier<T>` to Scala `() => T`. 2.12 and 2.13 works fine here. [Potential Solution](https://stackoverflow.com/questions/60410759/function-param-issue-in-scala-11-works-in-scala-12-using-resilience4j)

[the remark this answers is not in the document]
Comment (scottsand-db) [reply standalone/src/main/java/io/delta/standalone/VersionLog.java:44]: I don't see a solution to item 3.

So, if our choice comes down to 1) exposing this constructor, or 2) doing this in Scala, I would do 2.

btw, typically when you make a design doc and have different options, you try them out a bit until you know that each option is viable. so, if keeping the constructor package-private is not viable, then that might have changed our minds for the design doc. just friendly feedback :)

[the remark this answers is not in the document]
Comment (zsxwing) [reply standalone/src/main/java/io/delta/standalone/VersionLog.java:44]: hm, didn't realize this. This was the reason that the existing constructor is public. Then I agree that doing in scala is better.

**Comment by 6a0juu** [comment]
Closed. [Scala implementation](https://github.com/delta-io/connectors/pull/353) would be a stronger candidate because JAVA implementation has API exposing and Supplier type casting issues.

**Pull request closed without merging**